indico/modules/events/registration/api.py

```diff
@@ -38,34 +39,18 @@ class RHAPIRegistrant(RH):
 
     @oauth.require_oauth('registrants')
     def _checkParams(self):
-        event = ConferenceHolder().getById(request.view_args['event_id'], True)
-        if event is None:
+        self._event = ConferenceHolder().getById(request.view_args['event_id'], True)
+        if self._event is None:
             raise NotFound("No such event")
-        if not event.canManageRegistration(request.oauth.user):
+        if not self._event.canManageRegistration(request.oauth.user):
```

Comment:
should be in checkProtection and use `self.event.can_manage(request.oauth.user, role='registration')`